vectorstores/alloydb/options.go

```diff
@@ -0,0 +1,107 @@
+package alloydb
+
+import "context"
+
+const defaultPort = 5432 // default PostgreSQL port
+
+// Option is a function type that can be used to modify the Engine.
+type Option func(p *engineConfig)
+
+type engineConfig struct {
+	projectID       string
+	region          string
+	cluster         string
+	instance        string
+	host            string
+	port            int
+	database        string
+	user            string
+	password        string
+	ipType          string
+	iAmAccountEmail string
+	emailRetreiver  EmailRetreiver
+}
+
+// WithProjectID sets the ProjectId field.
+func WithProjectID(projectID string) Option {
+	return func(p *engineConfig) {
+		p.projectID = projectID
+	}
+}
+
+// WithRegion sets the Region field.
+func WithRegion(region string) Option {
+	return func(p *engineConfig) {
+		p.region = region
+	}
+}
+
+// WithCluster sets the Cluster field.
+func WithCluster(cluster string) Option {
+	return func(p *engineConfig) {
+		p.cluster = cluster
+	}
+}
+
+// WithInstance sets the Instance field.
+func WithInstance(instance string) Option {
+	return func(p *engineConfig) {
+		p.instance = instance
+	}
+}
+
+// WithHost sets the Host field.
+func WithHost(host string) Option {
+	return func(p *engineConfig) {
+		p.host = host
+	}
+}
+
+// WithPort sets the Port field.
+func WithPort(port int) Option {
+	return func(p *engineConfig) {
+		p.port = port
+	}
+}
```

Comment:
I think these are at odds with with project/region/cluster/instance. At a minimum I would suggest having logic that if these are specified and WithAlloyDBCluster is specified, it throws an error. (And probably logic that if neither are specified they should throw an error). 

But probably I would suggest supporting a "WithPool" over Host/Port -- users can specify their pool with whatever config they want. There is something similar to this in the python example